roles/tests-integration/lib/sniffer.rs

```diff
@@ -215,30 +244,56 @@ impl Sniffer {
         recv: Receiver<MessageFrame>,
         send: Sender<MessageFrame>,
         downstream_messages: MessagesAggregator,
-        intercept_messages: Vec<InterceptMessage>,
+        action: Option<Action>,
     ) -> Result<(), SnifferError> {
+        let mut blocked = false;
         while let Ok(mut frame) = recv.recv().await {
+            if blocked {
+                continue;
+            }
             let (msg_type, msg) = Self::message_from_frame(&mut frame);
-            let intercept_message = intercept_messages.iter().find(|im| {
-                im.direction == MessageDirection::ToUpstream && im.expected_message_type == msg_type
+            let action = action.as_ref().and_then(|action| match action {
```

Comment:
I would move this `match` into a specific function in `Action::fn_name(msgtype, msg_direction)` so we dont repeat it in the other function